src/NetherPortalScanner.cpp

```diff
@@ -297,15 +297,26 @@ void cNetherPortalScanner::OnDisabled(void)
 		Position.z += OutOffset;
 	}
 
+	auto EntityID = m_EntityID;
+	auto & DestinationWorld = m_World;
+	auto & SourceWorld = m_SourceWorld;
+	auto DestinationPosition = Position;
+
 	// Lookup our warping entity by ID
 	// Necessary as they may have been destroyed in the meantime (#4582)
-	m_SourceWorld.DoWithEntityByID(
-		m_EntityID,
-		[this, &Position](cEntity & a_Entity)
+	// And since this is called from the destination world's thread queue a task on the source world
+	m_SourceWorld.QueueTask(
+		[EntityID, &DestinationWorld, &SourceWorld, DestinationPosition](cWorld & a_World)
 		{
-			FLOGD("Placing player at {0}", Position);
-			a_Entity.MoveToWorld(m_World, Position, true, false);
-			return true;
+			SourceWorld.DoWithEntityByID(
```

Comment:
Nit: You don't need to capture `SourceWorld`
```suggestion
			a_World.DoWithEntityByID(
```

Reply:
good point